cmd/syncthing/main.go

```diff
@@ -728,6 +708,34 @@ func syncthingMain() {
 	cachedDiscovery := discover.NewCachingMux()
 	mainSvc.Add(cachedDiscovery)
 
+	// GUI
+
+	setupGUI(mainSvc, cfg, m, apiSub, cachedDiscovery, relaySvc, errors, systemLog)
+
+	// Initialize Folders
```

Comment:
I think there may be operations on the REST interface that panic or return funky results if called in this interval, i.e. to request a scan on a folder that doesn't exist and so on. At minimum this needs to be checked for.